cmd/exec_default.go

```diff
@@ -0,0 +1,52 @@
+// +build !linux,!darwin
+
+package cmd
+
+import (
+	"os"
+	osexec "os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+// exec executes the given command, passing it args and setting its environment
+// to env.
+// The exec function is allowed to never return and cause the program to exit.
+func exec(command string, args []string, env []string) error {
+	ecmd := osexec.Command(command, args...)
+	ecmd.Stdin = os.Stdin
+	ecmd.Stdout = os.Stdout
+	ecmd.Stderr = os.Stderr
+	ecmd.Env = env
+
+	signals := make([]os.Signal, 30)
+	for i := range signals {
+		signals[i] = syscall.Signal(i + 1)
+	}
+
+	// Forward SIGINT, SIGTERM, SIGKILL to the child command
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, signals...)
+
+	go func() {
+		sig := <-sigChan
+		if ecmd.Process != nil {
```

Comment:
There's a data race here between `ecmd.Run` and checking `ecmd.Process` here. See https://github.com/99designs/aws-vault/pull/162 for the fix; you want to call Start() and then Wait() instead of Run().